File: couch/ocaml/create/couch_create.ml
(* ------------------------------------------------------------------- *)
(*
	couch_create.ml

					Mar/14/2013

*)
(* ------------------------------------------------------------------- *)
open	Printf
open	Json_type
open	Curl_manipulate
(* ------------------------------------------------------------------- *)

let city_xx = Object ["t2021",Object ["name", String "長野";
                   "population", Int 51368; "date_mod", String "1995-5-21"];
		"t2022",Object ["name", String "松本";
                   "population", Int 58362; "date_mod", String "1995-9-30"];
		"t2023",Object ["name", String "上田";
                   "population", Int 82475; "date_mod", String "1995-4-12"];
		"t2024", Object ["name", String "小諸";
                   "population", Int 72369; "date_mod", String "1995-8-25"];
		"t2025", Object ["name", String "岡谷";
                   "population", Int 42953; "date_mod", String "1995-9-18"];
		"t2026", Object ["name", String "塩尻";
                   "population", Int 72648; "date_mod", String "1995-2-8"];
		"t2027", Object ["name", String "茅野";
                   "population", Int 62846; "date_mod", String "1995-4-14"];
		"t2028", Object ["name", String "飯田";
                   "population", Int 25719; "date_mod", String "1995-9-23"];
		"t2029", Object ["name", String "中野";
                   "population", Int 52987; "date_mod", String "1995-12-18"];
		"t2030", Object ["name", String "諏訪";
                   "population", Int 29856; "date_mod", String "1995-2-12"];
		"t2031", Object ["name", String "駒ヶ根";
                   "population", Int 32745; "date_mod", String "1995-4-15"];
		"t2032", Object ["name", String "佐久";
                   "population", Int 85972; "date_mod", String "1995-6-27"];
		"t2033", Object ["name", String "伊那";
                   "population", Int 52918; "date_mod", String "1995-7-25"];
		"t2034", Object ["name", String "千曲";
                   "population", Int 61479; "date_mod", String "1995-8-17"]]

(* ------------------------------------------------------------------- *)
let _ = print_endline "*** 開始 ***"
let url_collection = "http://host_dbase:5984/city"
let url_target = url_collection ^ "/cities"

let str_json = Json_io.string_of_json city_xx


let _ =
	let r,c = delete url_collection in
		printf "%d -> %s\n" r c

let _ =
	let r,c = put url_collection "" in
		printf "%d -> %s\n" r c

let _ =
	let r,c = put url_target str_json in
		printf "%d -> %s\n" r c

let _ = print_endline "*** 終了 ***"
(* ------------------------------------------------------------------- *)

File: couch/cplus/create/couch_create.cpp
// --------------------------------------------------------------------
/*
	couch_create.cpp

					Jun/19/2015
*/
// --------------------------------------------------------------------
#include	<iostream>
#include	<fstream>
#include	<sstream>
#include	<string>
#include	<map>

#include	<string.h>

#include "/var/www/data_base/common/cplus_common/include/picojson.h"

using namespace std;
using namespace picojson;

typedef map<string,string> Unit;
// --------------------------------------------------------------------
extern	map <string,Unit> dict_append_proc
	(map <string,Unit> dict_aa,string id,string name,
	int population,string date_mod);

extern void curl_put_proc (const char *url_target,string str_in);
extern void curl_delete_proc (const char url[]);
// --------------------------------------------------------------------
static map <string,Unit >  data_prepare_proc ()
{
	map <string,Unit> dict_aa;

	dict_aa = dict_append_proc (dict_aa,"t2021","長野",32197,"1922-6-12");

	dict_aa = dict_append_proc (dict_aa,"t2022","松本",97235,"1922-19-25");
	dict_aa = dict_append_proc (dict_aa,"t2023","上田",75128,"1922-3-21");
	dict_aa = dict_append_proc (dict_aa,"t2024","小諸",52486,"1922-9-8");

	dict_aa = dict_append_proc (dict_aa,"t2025","岡谷",83671,"1922-1-5");
	dict_aa = dict_append_proc (dict_aa,"t2026","塩尻",71948,"1922-5-22");
	dict_aa = dict_append_proc (dict_aa,"t2027","茅野",46185,"1922-7-17");
	dict_aa = dict_append_proc (dict_aa,"t2028","飯田",59614,"1922-3-04");
	dict_aa = dict_append_proc (dict_aa,"t2029","中野",71823,"1922-5-9");
	dict_aa = dict_append_proc (dict_aa,"t2030","諏訪",82361,"1922-8-12");
	dict_aa = dict_append_proc (dict_aa,"t2031","駒ヶ根",36428,"1922-5-25");
	dict_aa = dict_append_proc (dict_aa,"t2032","佐久",19854,"1922-4-8");
	dict_aa = dict_append_proc (dict_aa,"t2033","伊那",82617,"1922-5-17");
	dict_aa = dict_append_proc (dict_aa,"t2034","千曲",35284,"1922-6-21");

	return	dict_aa;
}
// --------------------------------------------------------------------
int main (int argc,char *argv[])
{
	cerr << "*** 開始 ***\n";

	char url_collection[] = "http://localhost:5984/nagano/";
	char url_target[160];

	curl_delete_proc (url_collection);
	curl_put_proc (url_collection,"");

	map <string,Unit> dict_aa = data_prepare_proc ();

	cout << "要素数：" << (unsigned int)dict_aa.size() << endl;


	map <string,Unit>:: iterator it = dict_aa.begin ();

	while (it != dict_aa.end ())
		{
		Unit unit_aa = (*it).second;
		string key = (*it).first;

		object oo;
		oo.insert (make_pair("name",unit_aa["name"]));
		oo.insert (make_pair("population",unit_aa["population"]));
		oo.insert (make_pair("date_mod",unit_aa["date_mod"]));

		value json_value_new = value (oo);
		string ssx = json_value_new.serialize ();

		strcpy (url_target,url_collection);
		strcat (url_target,key.c_str ());
		curl_put_proc (url_target,ssx);

		it++;
		}

	cerr << "*** 終了 ***\n";
}

// --------------------------------------------------------------------
